core/unit_test/TestExecSpacePartitioning.hpp

```diff
@@ -131,14 +131,14 @@ TEST(TEST_CATEGORY, partitioning_by_args) {
   auto instances =
       Kokkos::Experimental::partition_space(TEST_EXECSPACE(), 1, 1);
   ASSERT_EQ(int(instances.size()), 2);
-  test_partitioning(instances);
+  test_partitioning(instances[0], instances[1]);
 }
```

Comment:
Add a test which uses structured binding. Also lets add a
```
static_assert(std::is_same_v<decltype(instances), std::array<TEST_EXECSPACE, 2>>);
```
here and the corresponding one in the vector version.